testutils/consul.go

```diff
@@ -50,3 +53,95 @@ func NewTestConsulServer(tb testing.TB, config TestConsulServerConfig) *testutil
 
 	return srv
 }
+
+// Bulk add test data for seeding consul
+func AddServices(t testing.TB, srv *testutil.TestServer, svcs []testutil.TestService) {
+	for _, s := range svcs {
+		RegisterConsulService(t, srv, s, testutil.HealthPassing)
+	}
+}
+
+// Generate service TestService entries.
+// Services with different IDs and Names.
+func TestServices(n int) []testutil.TestService {
+	return generateServices(n,
+		func(i int) string {
+			return fmt.Sprintf("svc_name_%d", i)
+		},
+		func(i int) string {
+			return fmt.Sprintf("svc_id_%d", i)
+		})
+}
+
+// Generate service instance TestService entries.
+// Services with different IDs but with the same name.
+func TestInstances(n int) []testutil.TestService {
+	return generateServices(n,
+		func(i int) string {
+			return fmt.Sprintf("svc_name_common")
+		},
+		func(i int) string {
+			return fmt.Sprintf("svc_id_%d", i)
+		})
+}
+
+// shorter name for the formatting functions
+type fmtFunc func(i int) string
+
+// does the actual work of generating the TestService objects
+func generateServices(n int, namefmt, idfmt fmtFunc) []testutil.TestService {
+	baseport := 30000
+	services := make([]testutil.TestService, n)
+	for i := 0; i < n; i++ {
+		services[i] = testutil.TestService{
+			Name:    namefmt(i),
+			ID:      idfmt(i),
+			Address: "127.0.0.2",
+			Port:    int(baseport + i),
+			Tags:    []string{},
+		}
+	}
+	return services
+}
+
+// ---------------------------------------------------------------------------
+// The below functions are not used in tests but are handy to keep around for
+// when you need to check on Consul's data .
+
+// All registered services
+func ShowMeServices(t testing.TB, srv *testutil.TestServer) {
+	u := fmt.Sprintf("http://%s/v1/agent/services", srv.HTTPAddr)
+	resp := RequestHTTP(t, http.MethodGet, u, "")
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
For the checking errors in test functions, we've often been using `require.NoError` instead of the if-statement. I've been liking it since its a little cleaner to my eyes. Suggestion to use `require.NoError` if you'd like! If you do like, there are two more instances of error checks below (line 129 + 141)

Reply:
Good call. That is more compressed and fits better with the other code. I'll update those panics to NoError checks.